### PR TITLE
Handle bundle_upload_v2 id

### PR DESCRIPTION
**Pull request by pat-trunk-io**
Related to:
https://github.com/trunk-io/trunk/pull/19811

### CODE REVIEW

File: bundle/src/bundle_meta.rs [hunk 1]
```diff
@@ -100,6 +108,46 @@ pub struct BundleMetaV0_6_2 {
...
+#[cfg_attr(feature = "pyo3", gen_stub_pyclass, pyclass(get_all))]
+#[cfg_attr(feature = "wasm", derive(Tsify))]
+pub struct BundleMetaV0_6_3 {
+    #[serde(flatten)]
+    pub base_props: BundleMetaBaseProps,
+    #[serde(flatten)]
+    pub junit_props: BundleMetaJunitProps,
+    #[serde(flatten)]
+    pub debug_props: BundleMetaDebugProps,
+    pub bundle_upload_id_v2: Option<String>, // TODO(pat): make this required
```
Comment (TylerJang27) [inline bundle/src/bundle_meta.rs:121]: What's preventing us from making it required?

File: bundle/src/bundle_meta.rs
Comment (TylerJang27) [inline bundle/src/bundle_meta.rs:1]: Can you also modify `parse_meta_from_tarball` in `bundler.rs` and if time add a test case to `context-js/tests/parse_compressed_bundle.test.ts`?

**Review by TylerJang27 — COMMENTED** [review]
Missing parsing functionality

File: bundle/src/bundle_meta.rs (hunk 1 above)
Comment (pat-trunk-io) [reply bundle/src/bundle_meta.rs:121]: It wasn't rolled out to prod yet. Now it is so probably can make this required

Comment (pat-trunk-io) [reply bundle/src/bundle_meta.rs]: Done

File: bundle/src/bundle_meta.rs
Comment (pat-trunk-io) [reply bundle/src/bundle_meta.rs:1]: Is this good?

File: context-js/tests/parse_compressed_bundle.test.ts [hunk 2]
```diff
@@ -103,6 +103,14 @@ describe("context-js", () => {
         command_line: "trunk-analytics-cli upload --token=***",
       },
     ],
+    [
+      "V0_6_3",
+      {
+        num_tests: faker.number.int(100),
+        num_files: faker.number.int(100),
+        command_line: "trunk-analytics-cli upload --token=***",
```
Comment (TylerJang27) [inline context-js/tests/parse_compressed_bundle.test.ts:111]: I think this is missing `id_v2`? I'm surprised this compiled (`pnpm i --dir context-js; pnpm --dir context-js run build_and_test`)

**Review by TylerJang27 — APPROVED** [review]
1 sanity check

**Pull request merged**